src/plugins/controls/public/options_list/types.ts

```diff
@@ -7,26 +7,32 @@
  */
 
 import { ReduxEmbeddableState } from '@kbn/presentation-util-plugin/public';
+import { FieldSpec } from '@kbn/data-views-plugin/common';
+
 import { ControlOutput } from '../types';
 import {
-  OptionsListField,
   OptionsListSuggestions,
   OptionsListEmbeddableInput,
 } from '../../common/options_list/types';
 
+export const MIN_OPTIONS_LIST_REQUEST_SIZE = 10;
+export const MAX_OPTIONS_LIST_REQUEST_SIZE = 1000;
+
 interface SearchString {
   value: string;
   valid: boolean;
 }
 
 // Component state is only used by public components.
 export interface OptionsListComponentState {
-  field?: OptionsListField;
-  totalCardinality?: number;
   availableOptions?: OptionsListSuggestions;
+  allowExpensiveQueries: boolean;
   invalidSelections?: string[];
-  validSelections?: string[];
   searchString: SearchString;
+  validSelections?: string[];
+  totalCardinality?: number;
+  popoverOpen: boolean;
```

Comment:
I've made the status of the popover part of the component state so that I can ensure that it is closed both (a) when the component is unmounted and (b) when there is a fatal error (which is handled on the embeddable side).